tests/test_extract_model_langchain_openai.py

```diff
@@ -47,3 +49,254 @@ def test_entire_llm_call_using_langchain_openai(expected_model, model):
 
     generation = list(filter(lambda o: o.type == "GENERATION", trace.observations))[0]
     assert generation.model == expected_model
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(  # noqa: F821
+    "expected_model,model",
+    [
+        ("gpt-3.5-turbo", ChatOpenAI()),
+        ("gpt-3.5-turbo-instruct", OpenAI()),
+        (
+            "gpt-3.5-turbo",
+            AzureChatOpenAI(
+                openai_api_version="2023-05-15",
+                azure_deployment="your-deployment-name",
+                azure_endpoint="https://your-endpoint-name.azurewebsites.net",
+            ),
+        ),
+        (
+            "gpt-3.5-turbo-instruct",
+            AzureOpenAI(
+                openai_api_version="2023-05-15",
+                azure_deployment="your-deployment-name",
+                azure_endpoint="https://your-endpoint-name.azurewebsites.net",
+            ),
+        ),
+    ],
+)
+async def test_ainvoke_on_llm_call_using_langchain_openai(expected_model, model):
+    callback = CallbackHandler()
+    try:
+        # LLM calls are failing, because of missing API keys etc.
+        # However, we are still able to extract the model names beforehand.
```

Comment:
Do you have an openai key at hand? You can set the OPENAI_API_KEY env variable and we can test the entire LLM call and dont need the try..except block. This is also how we do it in our CI pipeline.